zerver/actions/presence.py

```diff
@@ -117,17 +123,16 @@ def do_update_user_presence(
     # times per minute with multiple connected browser windows.
     # We also need to be careful not to wrongly "update" the timestamp if we actually already
     # have newer presence than the reported log_time.
-    if (
-        not created
-        and log_time - presence.last_active_time > datetime.timedelta(seconds=55)
-        and log_time > presence.last_connected_time
+    if not created and log_time - presence.last_connected_time > datetime.timedelta(
+        seconds=settings.PRESENCE_UPDATE_MIN_FREQ_SECONDS
     ):
         presence.last_connected_time = log_time
         update_fields.append("last_connected_time")
     if (
         not created
         and status == UserPresence.LEGACY_STATUS_ACTIVE_INT
-        and log_time > presence.last_active_time
+        and time_since_last_active
+        > datetime.timedelta(seconds=settings.PRESENCE_UPDATE_MIN_FREQ_SECONDS)
     ):
         presence.last_active_time = log_time
         update_fields.append("last_active_time")
```

Comment:
Do we want to update `last_connected_time` whenever we're also updating `last_active_time`? Like, have a `if log_time > presence.last_connected_time: ` block here to update `last_connected_time` too?

I feel like that would make it easier to reason about things -- we could promise to downstream users of the system that `last_connected_time >= last_active_time`, which I think this implementation as written does not achieve.

Reply:
Good point, let's keep that property - updated. Would we want to have a db constraint for this to detect violations? I guess not worth it, but just throwing the idea - something like `alter table zerver_userpresence add constraint  X check (last_connected_time >= last_active_time);`

Reply:
I think given that we only write to the table in this one function, that is probably not necessary, but if it's easy to add via the Django constraint system, it might be a good way to catch bugs.